kasa/device_factory.py

```diff
@@ -132,7 +132,7 @@ def get_device_class_from_family(device_type: str) -> Optional[Type[SmartDevice]
         "SMART.TAPOPLUG": TapoPlug,
         "SMART.TAPOBULB": TapoBulb,
         "SMART.KASAPLUG": TapoPlug,
-        "SMART.KASASWITCH": TapoBulb,
+        "SMART.KASASWITCH": TapoPlug,
```

Comment:
To my understanding this would break the brightness functionality, so I'd wait until we get a more modular base class for the SMART devices.

Reply:
Reverted